lib/engine/game/g_system18/map_britain_customization.rb

```diff
@@ -0,0 +1,497 @@
+          !corporation.ipoed && entity.companies.find { |c| c.sym == corporation.name }
+        end
+
+        def map_britain_after_par(corporation)
+          entity = corporation.owner
+
+          company = entity.companies.find { |c| c.sym == corporation.name }
+          raise GameError, 'Logic error, no matching company found' unless company
+
+          entity.companies.delete(company)
+          company.close!
+        end
+
+        # can the corporation reach an icon?
+        def map_britain_can_remove_icon?(entity)
+          return false unless entity.corporation?
+          return false if @britain_corps_with_mines[entity.name]
+
+          @britain_mines.any? { |m| @graph.reachable_hexes(entity).include?(m) }
+        end
+
+        def map_britain_icon_hexes(entity)
+          return [] unless entity.corporation?
+          return [] if @britain_corps_with_mines[entity.name]
+
+          @britain_mines.select { |m| @graph.reachable_hexes(entity).include?(m) }.map(&:id)
+        end
+
+        def map_britain_remove_icon(entity, hex_id)
+          return unless entity.corporation?
+          return if @britain_corps_with_mines[entity.name]
+
+          hex = hex_by_id(hex_id)
+          hex.tile.icons.clear # should be the only icon on the tile
+          @log << "#{entity.name} takes mine token from #{hex_id}"
+          @britain_corps_with_mines[entity.name] = hex_id
+        end
+
+        def map_britain_removable_icon_action_str
+          'Take a mine token'
+        end
+
+        def map_britain_extra_revenue_for(route, stops)
+          map_britain_bonuses(route, stops)[:revenue]
+        end
+
+        def map_britain_extra_revenue_str(route)
+          bonus = map_britain_bonuses(route, route.stops)[:description]
+          bonus ? " + #{bonus}" : ''
+        end
+
+        def map_britain_bonuses(route, stops)
+          train = route.train
+          bonus = { revenue: 0 }
+          return bonus unless train
+
+          desc = ''
+
+          # mines (no 4D doubling), only on one run
+          corp = train.owner
+          lowest_train = route.routes.reject { |r| r.stops.empty? }.min_by { |r| corp.trains.index(r.train) }.train
+          if (train == lowest_train) && corp && @britain_corps_with_mines[corp.name]
+            bonus[:revenue] += BRITAIN_MINE_BONUS[@phase.name.to_i - 2]
+            desc = 'Mine'
+          end
+
+          # Scotland (doubles with 4D)
+          scotland_city = stops.find do |stop|
+            BRITAIN_REGION_HEXES['Scotland'].include?(stop.hex.id) && (stop.city? || stop.offboard?)
+          end
+          non_scotland_city = stops.find { |stop| !BRITAIN_REGION_HEXES['Scotland'].include?(stop.hex.id) && stop.city? }
+          if scotland_city && non_scotland_city
+            bonus[:revenue] += BRITAIN_SCOTLAND_BONUS[@phase.name.to_i - 2] * (train.name == '4D' ? 2 : 1)
+            desc += ', ' unless desc == ''
+            desc += 'Scotland'
+          end
```

Comment:
If you've created the `@scotland_hexes` variable, you could simplify this test by using set differences.
```suggestion
          stop_hexes = stops.map(&:hex)
          if stop_hexes.intersect?(@scotland_hexes) && !stop_hexes.difference(@scotland_hexes).empty?
            bonus[:revenue] += BRITAIN_SCOTLAND_BONUS[@phase.name.to_i - 2] * (train.name == '4D' ? 2 : 1)
            desc += ', ' unless desc == ''
            desc += 'Scotland'
          end
```

Reply:
Style.